File: 1642-water-bottles/water-bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int empty =0, full=numBottles;
        int sol=0;

        while(full!=0){
            sol+=full;
            empty+=full;
            full = empty/numExchange;
            empty %= numExchange;
        }

        return sol;
    }
};

File: 1642-water-bottles/water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        total = numBottles
        empty = numBottles

        while(empty>=numExchange):
            new = empty//numExchange
            empty=empty%numExchange + new
            total+=new

        return total
